src/features/SingleProductPage.js

```diff
@@ -1,14 +1,78 @@
 import React from "react";
 import { useParams } from "react-router-dom";
 import { useGetProductQuery } from "./api/apiSlice";
+import { useState, useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { productQuantitySetTo } from "./cart/cartSlice";
 
 export default function SingleProductPage() {
   const { productId } = useParams();
+  const dispatch = useDispatch();
   const { data: product, isLoading } = useGetProductQuery(productId);
+  const [amountOfItems, setAmountOfItems] = useState(1);
+  const cart = useSelector((state) => state.cart);
+
+  useEffect(() => {
+    if (product) {
+      const parsedProductId = parseInt(productId);
+      const productInCart = cart.find(
+        (cartItem) => cartItem.id === parsedProductId
+      );
+      if (productInCart) {
+        const productQuantity = productInCart.quantity;
+        setAmountOfItems(productQuantity);
+      }
+    }
+  }, [product, productId, cart]);
+
+  function onAddItemsToCart(event) {
+    event.preventDefault();
+    const updatedProduct = { ...product, quantity: amountOfItems };
```

Comment:
also more a matter of taste:
since onAddItemsToCart is a form submitHandler it would be a bit clearer to get the amountOfItems from event.target.amount.value (=the input value) instead of the state variable. both should always be the same, still the function is 'purer' to read if it is not capturing th eamountOfItems variable from its outer scope.

Reply:
done.